ckanext/datajson/harvester_base.py

```diff
@@ -260,20 +267,22 @@ def gather_stage(self, harvest_job):
             if not matched_filters:
                 continue
 
-            if parent_identifiers and new_parents \
-                and dataset['identifier'] not in parent_identifiers \
-                and dataset.get('isPartOf') in new_parents:
-                if run_status == 'parents_run':
-                    # skip those whose parents still need to run.
-                    continue
-                else:
-                    # which is 'children_run'.
-                    # error out since parents got issues.
-                    self._save_gather_error(
-                        "Record with identifier '%s': isPartOf '%s' points to \
-                        an erroneous record." % (dataset['identifier'],
-                            dataset.get('isPartOf')), harvest_job)
-                    continue
+            # old code for previous catalog
+            if p.toolkit.check_ckan_version(max_version='2.7.99'):
```

Comment:
I wasn't a fan of this code before, but now intermingling the CKAN 2.8 checks make this even harder to reason about.

Instead of doing it like this, let's create a new harvest class DatajsonHarvester23 for the old CKAN 2.3 code and DatajsonHarvester for the new code. Then you only need to do the version check once to figure out which implementation to use and the readability goes way up. Not to mention you can write a very different algorithm without worrying about CKAN 2.3.

Reply:
I cleaned almost all the `check_ckan_version` calls in this file. This one now is the **only** one we use.
Do you think we still need this `DatajsonHarvester23` harvester @adborden?

Reply:
Yes, we should still split the implementation at the class level because the tests are still very hard to reason about.